### PR TITLE
unix: harden string copying, introduce strscpy()

### PR DESCRIPTION
**Pull request by bnoordhuis**
Replace calls to strcpy() and strncpy() with the newly introduced
uv__strscpy() function that is meticulous about zero-terminating
the destination buffer.

Refs: https://www.kernel.org/doc/htmldocs/kernel-api/API-strscpy.html
CI: https://ci.nodejs.org/job/libuv-test-commit/1091/

### CODE REVIEW

**Comment by kristensp** [comment]
Add

**Comment by refack** [comment]
Nit: why the `unix:` in the title? IIUC it's used on windows as well via `UV_ERR_NAME_GEN_R`

**Comment by bnoordhuis** [comment]
Because there were no strcpy/strncpy calls in src/win that I could find. (There are a few wcscpy/wcsncpy calls but I didn't touch those; something for another day.)

You're right the one in src/uv-common.c is win32+unix but the bulk of this PR is unix only.

File: src/unix/aix.c [hunk 1]
```diff
@@ -358,19 +358,15 @@ void uv_loadavg(double avg[3]) {
...
   if (dp == 0)
     return 0;
 
-  *dp = 0;
-  strcpy(rawbuf, cp);
-  *dp = '/';
-  strcat(rawbuf, "/r");
-  strcat(rawbuf, dp+1);
-  return rawbuf;
+  snprintf(*dst, sizeof(*dst), "%.*s/r/%s", (int) (cp - dp), cp, dp + 1);
```
Comment (richardlau) [inline src/unix/aix.c]: Kind of late/early morning for me, but shouldn't this be
```suggestion
  snprintf(*dst, sizeof(*dst), "%.*s/r%s", (int) (dp - cp), cp, dp + 1);
```
? I think `dp` should be `> cp` so (`cp - dp` would be negative).

**Comment by bnoordhuis** [comment]
Included feedback and rebased against v1.x HEAD: https://ci.nodejs.org/job/libuv-test-commit/1120/

**Comment by bnoordhuis** [comment]
Once more to see if the macos11 failure is a flake: https://ci.nodejs.org/job/libuv-test-commit/1121/

**Pull request closed without merging**